pkg/controller/podgc/gc_controller_test.go

```diff
@@ -84,8 +84,8 @@ func TestGCTerminated(t *testing.T) {
 				{name: "c", phase: v1.PodFailed},
 			},
 			threshold:                     1,
-			patchedPodNames:               sets.NewString(),
-			deletedPodNames:               sets.NewString("a", "b"),
+			patchedPodNames:               sets.New[string](),
+			deletedPodNames:               sets.New[string]("a", "b"),
```

Comment:
```suggestion
			deletedPodNames:               sets.New("a", "b"),
```

And then we can add `cmp.EquateEmpty` to match nil with empty set.

Reply:
sgtm, done